simple.cs

```diff
@@ -0,0 +1,46 @@
+﻿
+// to compile:
+// csc.exe -o+ simple.cs
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System.Linq;
+
+namespace freq
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Dictionary<string, int> tbl = new Dictionary<string, int>(StringComparer.Ordinal);
+            int value;
+
+
+            string line;
+            string[] words;
+            while ((line = Console.ReadLine()) != null)
+            {
+                words = line.Split((char[])null);
+                foreach (string w in words)
+                {
+                    if(w.Length == 0) {
+                        continue;
+                    }
+                    string lower = w.ToLower();
+                    tbl[lower] = (tbl.TryGetValue(lower, out value)) ? tbl[lower] = value + 1 : tbl[lower] = 1;
+                }
+            }
```

Comment:
I'm not a C# programmer, but it seems to me there are a few things that aren't particularly simple (and hence likely not idiomatic) here. You can use `StringSplitOptions.RemoveEmptyEntries` on the `Split()` call to avoid the `w.Length` check. You can use `Dictionary.GetValueOrDefault` to simplify the inner loop. Also, it's a bit better more efficient (and no less simple) to lowercase the entire line at once, rather than per word.

What about the following (also tweaked a couple of the names):

```c#
        var counts = new Dictionary<string, int>();
        string line;
        while ((line = Console.ReadLine()) != null)
        {
            line = line.ToLower();
            var words = line.Split(' ', StringSplitOptions.RemoveEmptyEntries);
            foreach (string word in words)
            {
                counts[word] = counts.GetValueOrDefault(word, 0) + 1;
            }
        }
        var ordered = counts.OrderByDescending(pair => pair.Value);
        foreach (var entry in ordered)
        {
            Console.WriteLine("{0} {1}", entry.Key, entry.Value);
        }
```

Reply:
That looks better!